File: C#/0198. House Robber.cs
public class Solution {
    public int Rob(int[] nums) {
        int lenNums=nums.Length;
        if(lenNums==0){
            return 0;
        }
        if(lenNums==1){
            return nums[0];
        }
        IList<int> dp=new List<int>();
        dp.Add(0);
        dp.Add(nums[0]);
        for(int i=2;i<lenNums+1;i++){
            dp.Add(Math.Max(dp[i-1],dp[i-2]+nums[i-1]));
        }
        return Math.Max(dp[lenNums],dp[lenNums-1]);
    }
}

File: archives/20190519python/0198. House Robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if not nums:
            return 0
        if len(nums)==1:
            return nums[0]
        dp=[0,nums[0]]
        for num in nums[1:]:
            dp.append(max(dp[-1],dp[-2]+num))
        return dp[-1]
